src/main/java/seedu/address/ui/PropertyCard.java

```diff
@@ -58,9 +68,21 @@ public PropertyCard(Property property, int displayedIndex) {
         phone.setText(property.getSeller().getPhone().value);
         address.setText(property.getAddress().value);
         email.setText(property.getSeller().getEmail().value);
-        property.getTags().stream()
-                .sorted(Comparator.comparing(tag -> tag.tagName))
-                .forEach(tag -> tags.getChildren().add(new Label(tag.tagName)));
+
+        // Allow wrapping of labels
+        ObservableDoubleValue maxWidth = parentWidthProperty.subtract(WIDTH_PADDING);
+        phone.maxWidthProperty().bind(maxWidth);
+        seller.maxWidthProperty().bind(maxWidth);
+        email.maxWidthProperty().bind(maxWidth);
+
+        List<Tag> tagList = new ArrayList<>(property.getTags());
+        tagList.sort(Comparator.comparing(tag -> tag.tagName));
+        for (Tag tag : tagList) {
+            Label label = new Label(tag.tagName);
+            label.setWrapText(true);
+            label.maxWidthProperty().bind(maxWidth);
+            tags.getChildren().add(label);
+        }
```

Comment:
Can we improve the SLAP here? We can consider abstracting this entire function out.

Reply:
will leave this to the future